### PR TITLE
Watch later

### PR DESCRIPTION
**Pull request by krate27**

### CODE REVIEW

**Review by hinanshisuthar — COMMENTED** [review]
Nice work done implementing the logic. Clean code practices followed. Good to megre.

File: src/Components/Sidebar/sidebar.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            id="btn"
+            onClick={() => setSideBar((sideBar) => !sideBar)}
+          ></i>
+        </div>
+        <ul className="nav-list">
+          <li>
+            <a href="#">
+              <i class="fa-solid fa-heart"></i>
+              <span className="links_name">Liked</span>
+            </a>
```
Comment (partha8) [inline src/Components/Sidebar/sidebar.js:24]: You'll add routes here in the future right?.. I'm talking about the anchor tags here.

File: src/Context/dataContext.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    //   try {
+    //     const vidData = await getVideoService();
+    //     Datadispatch({
+    //       type: Abbreviations.INITIAL_VIDEOS,
+    //       payload: vidData.data.videos,
+    //     });
+    //   } catch (error) {
+    //     console.log(error);
+    //   }
+    // })();
```
Comment (partha8) [inline src/Context/dataContext.js:26]: remove this if not needed

File: src/custom-hooks/useFetch.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    (async () => {
+      try {
+        const { data, status } = await axios.get(url, { signal: abortCont });
+        if (status === 200) {
+          setData(data[dataName]); //data[videos]
+          setLoader(false);
+        }
+      } catch (error) {
+        if (error.name === "AbortError") {
+          console.log("fetch aborted");
```
Comment (partha8) [inline src/custom-hooks/useFetch.js:21]: Have an alert/toast here too.

File: src/Components/Sidebar/sidebar.js (hunk 1 above)
Comment (krate27) [reply src/Components/Sidebar/sidebar.js:24]: yes

File: src/Context/dataContext.js (hunk 2 above)
Comment (krate27) [reply src/Context/dataContext.js:26]: I will do so finally, as it branch, so trying stuffs.

File: src/custom-hooks/useFetch.js (hunk 3 above)
Comment (krate27) [reply src/custom-hooks/useFetch.js:21]: will add in last

**Pull request merged**